src/minisweagent/run/extra/swebench_eval.py

```diff
@@ -0,0 +1,187 @@
+    # diff --git a/django/db/models/sql/.backup.query.py b/django/db/models/sql/.backup.query.py
+    # new file mode 100644
+    # index 0000000000..fc13db5948
+
+    # We "find" the first line that starts with "diff" and then we remove lines before it
+    lines = patch.split("\n")
+    for i, line in enumerate(lines):
+        if line.startswith("diff --git"):
+            patch = "\n".join(lines[i:])
+            break
+
+    return patch.rstrip() + "\n"  # Make sure the last line ends with a newline
+
+
+def evaluate_result(instance, model_patch, instance_id, dataset, sweagent_config) -> tuple[dict[str, Any], str]:
+    """Apply patch and evaluate the solution."""
+    from swebench.harness.constants import (
+        APPLY_PATCH_FAIL,
+        APPLY_PATCH_PASS,
+    )
+    from swebench.harness.grading import get_eval_report
+    from swebench.harness.test_spec.test_spec import make_test_spec
+
+    if not model_patch:
+        return {}, f"No git patch found for instance {instance_id}"
+
+    env = None
+    try:
+        sweagent_config.setdefault("environment", {})
+        sweagent_config["environment"]["writeable_tmp"] = True
+        env = get_sb_environment(sweagent_config, instance)
+        assert isinstance(env, SingularityEnvironment), "SWEBench Evaluation only supports Singularity environment"
+        assert hasattr(env, "sandbox_dir"), "expected Singularity Env with 'sandbox_dir' attribute"
+    except Exception as e:
+        return {}, f"Env creation failed with {e}"
+
+    test_spec = make_test_spec(instance=instance)
+    model_patch = process_git_patch(model_patch)
+
+    # Get patch and save it to /tmp/patch.diff
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir = Path(temp_dir)
+        # Patch file
+        patch_file_path = temp_dir / "patch.diff"
+        with open(patch_file_path, "w") as f:
+            f.write(model_patch)
+        dst = env.sandbox_dir / "tmp"
+        shutil.copy(patch_file_path, dst)
+        # Eval script
+        eval_script_path = temp_dir / "eval.sh"
+        with open(eval_script_path, "w") as f:
+            f.write(test_spec.eval_script)
+        shutil.copy(eval_script_path, dst)
+
+    # Set +x
+    logger.info("chmod /tmp/eval.sh", extra={"msg_type": "ACTION"})
+    obs = env.execute("chmod +x /tmp/eval.sh", cwd="/")
+    assert obs["returncode"] == 0, (
+        f"Got bad observation: {obs} while executing `chmod +x /tmp/eval.sh` for environment dir: {env.sandbox_dir}"
+    )
+
+    # Apply patch
+    # TODO (sumanthrh): This is from SkyAgent: https://github.com/NovaSky-AI/SkyRL/tree/main/skyagent, should probably just match the eval function in SWE-Bench repo
+    exec_command = (
+        "cd /testbed && "
+        "(git apply -v /tmp/patch.diff && echo 'APPLY_PATCH_PASS' || "
```

Comment:
We should probably be a lot more opinionated here, only support patches that apply cleanly with `git apply`, execute it with `.execute` and check if the return code is 0 to see if it was successful.

I realize this will not be compatible with the swebench eval and we should not claim it is. It will just be an eval that can evaluate mini-swe-agent traces well and be integrated into an RL setup. The final eval on the quality of the policy would still need to be done with the swebench harness.